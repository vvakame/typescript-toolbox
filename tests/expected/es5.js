var Sample = (function () {
    function Sample() {
    }
    Object.defineProperty(Sample.prototype, "name", {
        get: function () {
            return this._name;
        },
        set: function (name) {
            this._name = name;
        },
        enumerable: true,
        configurable: true
    });
    return Sample;
})();
