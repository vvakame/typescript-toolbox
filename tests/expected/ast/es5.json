{
  "_start": 0,
  "_end": 137,
  "_trailingTriviaWidth": 0,
  "_astID": 259,
  "_preComments": null,
  "_postComments": null,
  "moduleElements": {
    "_start": 0,
    "_end": 136,
    "_trailingTriviaWidth": 1,
    "_astID": 258,
    "_preComments": null,
    "_postComments": null,
    "_fileName": "tmp.ts",
    "members": [
      {
        "_start": 0,
        "_end": 136,
        "_trailingTriviaWidth": 1,
        "_astID": 257,
        "_preComments": null,
        "_postComments": null,
        "modifiers": [],
        "identifier": {
          "_start": 6,
          "_end": 12,
          "_trailingTriviaWidth": 0,
          "_astID": 221,
          "_preComments": null,
          "_postComments": null,
          "_text": "Sample",
          "_valueText": null
        },
        "typeParameterList": null,
        "heritageClauses": {
          "_start": 13,
          "_end": 13,
          "_trailingTriviaWidth": 0,
          "_astID": 222,
          "_preComments": null,
          "_postComments": null,
          "_fileName": "tmp.ts",
          "members": []
        },
        "classElements": {
          "_start": 16,
          "_end": 134,
          "_trailingTriviaWidth": 1,
          "_astID": 256,
          "_preComments": null,
          "_postComments": null,
          "_fileName": "tmp.ts",
          "members": [
            {
              "_start": 16,
              "_end": 38,
              "_trailingTriviaWidth": 1,
              "_astID": 227,
              "_preComments": null,
              "_postComments": null,
              "modifiers": [
                2
              ],
              "variableDeclarator": {
                "_start": 24,
                "_end": 37,
                "_trailingTriviaWidth": 0,
                "_astID": 226,
                "_preComments": null,
                "_postComments": null,
                "propertyName": {
                  "_start": 24,
                  "_end": 29,
                  "_trailingTriviaWidth": 0,
                  "_astID": 223,
                  "_preComments": null,
                  "_postComments": null,
                  "_text": "_name",
                  "_valueText": null
                },
                "typeAnnotation": {
                  "_start": 29,
                  "_end": 37,
                  "_trailingTriviaWidth": 0,
                  "_astID": 225,
                  "_preComments": null,
                  "_postComments": null,
                  "type": {
                    "_start": 31,
                    "_end": 37,
                    "_trailingTriviaWidth": 0,
                    "_astID": 224,
                    "_preComments": null,
                    "_postComments": null,
                    "_nodeType": 69,
                    "_text": "string",
                    "_valueText": "string"
                  }
                },
                "equalsValueClause": null
              }
            },
            {
              "_start": 40,
              "_end": 84,
              "_trailingTriviaWidth": 1,
              "_astID": 239,
              "_preComments": null,
              "_postComments": null,
              "modifiers": [],
              "propertyName": {
                "_start": 44,
                "_end": 48,
                "_trailingTriviaWidth": 0,
                "_astID": 228,
                "_preComments": null,
                "_postComments": null,
                "_text": "name",
                "_valueText": null
              },
              "parameterList": {
                "_start": 48,
                "_end": 50,
                "_trailingTriviaWidth": 0,
                "_astID": 230,
                "_preComments": null,
                "_postComments": null,
                "openParenTrailingComments": null,
                "parameters": {
                  "_start": 49,
                  "_end": 49,
                  "_trailingTriviaWidth": 0,
                  "_astID": 229,
                  "_preComments": null,
                  "_postComments": null,
                  "_fileName": "tmp.ts",
                  "members": [],
                  "_separatorCount": 0
                }
              },
              "typeAnnotation": {
                "_start": 50,
                "_end": 58,
                "_trailingTriviaWidth": 1,
                "_astID": 232,
                "_preComments": null,
                "_postComments": null,
                "type": {
                  "_start": 52,
                  "_end": 58,
                  "_trailingTriviaWidth": 0,
                  "_astID": 231,
                  "_preComments": null,
                  "_postComments": null,
                  "_nodeType": 69,
                  "_text": "string",
                  "_valueText": "string"
                }
              },
              "block": {
                "_start": 59,
                "_end": 84,
                "_trailingTriviaWidth": 1,
                "_astID": 238,
                "_preComments": null,
                "_postComments": null,
                "statements": {
                  "_start": 63,
                  "_end": 81,
                  "_trailingTriviaWidth": 1,
                  "_astID": 237,
                  "_preComments": null,
                  "_postComments": null,
                  "_fileName": "tmp.ts",
                  "members": [
                    {
                      "_start": 63,
                      "_end": 81,
                      "_trailingTriviaWidth": 1,
                      "_astID": 236,
                      "_preComments": null,
                      "_postComments": null,
                      "expression": {
                        "_start": 70,
                        "_end": 80,
                        "_trailingTriviaWidth": 0,
                        "_astID": 235,
                        "_preComments": null,
                        "_postComments": null,
                        "expression": {
                          "_start": 70,
                          "_end": 74,
                          "_trailingTriviaWidth": 0,
                          "_astID": 233,
                          "_preComments": null,
                          "_postComments": null,
                          "_text": "this",
                          "_valueText": "this"
                        },
                        "name": {
                          "_start": 75,
                          "_end": 80,
                          "_trailingTriviaWidth": 0,
                          "_astID": 234,
                          "_preComments": null,
                          "_postComments": null,
                          "_text": "_name",
                          "_valueText": null
                        }
                      }
                    }
                  ]
                },
                "closeBraceLeadingComments": null,
                "closeBraceToken": {
                  "_start": 83,
                  "_end": 84
                }
              }
            },
            {
              "_start": 86,
              "_end": 134,
              "_trailingTriviaWidth": 1,
              "_astID": 255,
              "_preComments": null,
              "_postComments": null,
              "modifiers": [],
              "propertyName": {
                "_start": 90,
                "_end": 94,
                "_trailingTriviaWidth": 0,
                "_astID": 240,
                "_preComments": null,
                "_postComments": null,
                "_text": "name",
                "_valueText": null
              },
              "parameterList": {
                "_start": 94,
                "_end": 108,
                "_trailingTriviaWidth": 1,
                "_astID": 246,
                "_preComments": null,
                "_postComments": null,
                "openParenTrailingComments": null,
                "parameters": {
                  "_start": 95,
                  "_end": 107,
                  "_trailingTriviaWidth": 0,
                  "_astID": 245,
                  "_preComments": null,
                  "_postComments": null,
                  "_fileName": "tmp.ts",
                  "members": [
                    {
                      "_start": 95,
                      "_end": 107,
                      "_trailingTriviaWidth": 0,
                      "_astID": 244,
                      "_preComments": null,
                      "_postComments": null,
                      "dotDotDotToken": null,
                      "modifiers": [],
                      "identifier": {
                        "_start": 95,
                        "_end": 99,
                        "_trailingTriviaWidth": 0,
                        "_astID": 241,
                        "_preComments": null,
                        "_postComments": null,
                        "_text": "name",
                        "_valueText": null
                      },
                      "questionToken": null,
                      "typeAnnotation": {
                        "_start": 99,
                        "_end": 107,
                        "_trailingTriviaWidth": 0,
                        "_astID": 243,
                        "_preComments": null,
                        "_postComments": null,
                        "type": {
                          "_start": 101,
                          "_end": 107,
                          "_trailingTriviaWidth": 0,
                          "_astID": 242,
                          "_preComments": null,
                          "_postComments": null,
                          "_nodeType": 69,
                          "_text": "string",
                          "_valueText": "string"
                        }
                      },
                      "equalsValueClause": null
                    }
                  ],
                  "_separatorCount": 0
                }
              },
              "block": {
                "_start": 109,
                "_end": 134,
                "_trailingTriviaWidth": 1,
                "_astID": 254,
                "_preComments": null,
                "_postComments": null,
                "statements": {
                  "_start": 113,
                  "_end": 131,
                  "_trailingTriviaWidth": 1,
                  "_astID": 253,
                  "_preComments": null,
                  "_postComments": null,
                  "_fileName": "tmp.ts",
                  "members": [
                    {
                      "_start": 113,
                      "_end": 131,
                      "_trailingTriviaWidth": 1,
                      "_astID": 252,
                      "_preComments": null,
                      "_postComments": null,
                      "expression": {
                        "_start": 113,
                        "_end": 130,
                        "_trailingTriviaWidth": 0,
                        "_astID": 251,
                        "_preComments": null,
                        "_postComments": null,
                        "_nodeType": 174,
                        "left": {
                          "_start": 113,
                          "_end": 123,
                          "_trailingTriviaWidth": 1,
                          "_astID": 249,
                          "_preComments": null,
                          "_postComments": null,
                          "expression": {
                            "_start": 113,
                            "_end": 117,
                            "_trailingTriviaWidth": 0,
                            "_astID": 247,
                            "_preComments": null,
                            "_postComments": null,
                            "_text": "this",
                            "_valueText": "this"
                          },
                          "name": {
                            "_start": 118,
                            "_end": 123,
                            "_trailingTriviaWidth": 0,
                            "_astID": 248,
                            "_preComments": null,
                            "_postComments": null,
                            "_text": "_name",
                            "_valueText": null
                          }
                        },
                        "right": {
                          "_start": 126,
                          "_end": 130,
                          "_trailingTriviaWidth": 0,
                          "_astID": 250,
                          "_preComments": null,
                          "_postComments": null,
                          "_text": "name",
                          "_valueText": null
                        }
                      }
                    }
                  ]
                },
                "closeBraceLeadingComments": null,
                "closeBraceToken": {
                  "_start": 133,
                  "_end": 134
                }
              }
            }
          ]
        },
        "closeBraceToken": {
          "_start": 135,
          "_end": 136
        }
      }
    ]
  },
  "endOfFileTokenLeadingComments": null,
  "_fileName": "tmp.ts"
}